packages/paste-website/src/utils/color-combos/index.ts

```diff
@@ -0,0 +1,159 @@
+import * as lodash from 'lodash';
+
+// Traditional import as the color package isn't exported and typed correctly
+const Color = require('color');
+
+interface Color {
+  color: number[];
+  model: string;
+  valpha: number;
+  hex: () => string;
+  contrast: (color: Color) => number;
+}
+
+export interface ColorCombinationAccessibility {
+  aa: boolean;
+  aaLarge: boolean;
+  aaa: boolean;
+  aaaLarge: boolean;
+}
+
+export interface ColorCombinationTypes {
+  accessibility: ColorCombinationAccessibility;
+  color?: number[];
+  contrast: number;
+  hex: string;
+  model?: string;
+  valpha?: number;
+}
+
+export interface ColorCombosTypes {
+  color?: number[];
+  combinations: ColorCombinationTypes[];
+  hex: string;
+  model?: string;
+  valpha?: number;
+  [key: number]: ColorCombosTypes;
+}
+
+interface Options {
+  threshold?: number;
+  compact?: boolean;
+  uniq?: boolean;
+}
+
+const ColorCombos = (
+  colors: string[] | {[name: string]: string},
+  options: Options = {}
+): ColorCombosTypes[] | false => {
+  let arr: Color[] = [];
+  let results: ColorCombosTypes[] = [];
+
+  const MINIMUMS: {aa: number; aaLarge: number; aaa: number; aaaLarge: number} = {
+    aa: 4.5,
```

Comment:
How did you settle on these values?

Reply:
This is the very wordy version https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html